src/components/views/rooms/AudioRecorder.js

```diff
@@ -0,0 +1,111 @@
+//@flow
+
+import React from 'react';
+import classNames from 'classnames';
+
+import sdk from '../../../index';
+import Modal from '../../../Modal';
+import ErrorDialog from '../dialogs/ErrorDialog';
+
+export default class AudioRecorder extends React.Component {
+  static propTypes = {
+    sendAudioBlob: React.PropTypes.func.isRequired,
+  };
+
+  mediaRecorder: ?MediaRecorder = null;
+  blobs: Array<Blob> = [];
+
+  constructor(props: Object) {
+    super(props);
+
+    this.state = {
+      recording: false,
+    };
+  }
+
+  onRecordClicked = async () => {
+    if(this.state.recording) {
+      this.mediaRecorder.stop();
+      this.mediaRecorder = null;
+      this.mediaStream.getTracks().forEach((track) => track.stop());
+      this.mediaStream = null;
+    } else {
+      if(!navigator.mediaDevices || !MediaRecorder) {
+        Modal.createDialog(ErrorDialog, {
+          description: `Sorry, but the features required for push 
+ to talk messages are not supported by your browser`,
```

Comment:
I think ErrorDialog requires title, button and focus: https://github.com/matrix-org/matrix-react-sdk/blob/master/src/components/views/dialogs/ErrorDialog.js#L33